bokeh/core/_templates/doc_js.js

```diff
@@ -1,4 +1,25 @@
-var docs_json = {{ docs_json }};
-var render_items = {{ render_items }};
+(function(root) {
+  function embed_document(root) {
+    var docs_json = {{ docs_json }};
+    var render_items = {{ render_items }};
 
-Bokeh.embed.embed_items(docs_json, render_items{%- if app_path -%}, "{{ app_path }}" {%- endif -%}{%- if absolute_url -%}, "{{ absolute_url }}" {%- endif -%});
+    root.Bokeh.embed.embed_items(docs_json, render_items{%- if app_path -%}, "{{ app_path }}" {%- endif -%}{%- if absolute_url -%}, "{{ absolute_url }}" {%- endif -%});
+  }
+
+  if (root.Bokeh !== undefined) {
+    embed_document(root);
+  } else {
+    var attempts = 0;
+    var timer = setInterval(function(root) {
+      if (root.Bokeh !== undefined) {
+        embed_document(root);
+        clearInterval(timer);
+      }
+      attempts++;
+      if (attempts > 10) {
+        console.log("Bokeh: ERROR: Unable to embed document because BokehJS library is missing")
+        clearInterval(timer);
+      }
+    }, 100, root)
```

Comment:
I still think 10ms interval with 100 tries is better

Reply:
In my subjective view, 100ms is already a visible delay. Given that the test is cheap, then I would go for more fine-grained intervals.

Reply:
cc @canavandl

Reply:
Fixed